### PR TITLE
Display image on SD card error

### PR DESCRIPTION
**Pull request by kamotswind**
This displays one of two error images if there is a fatal SD card error.

### CODE REVIEW

File: gnuboy-go/main/main.c [hunk 1]
```diff
@@ -251,7 +251,11 @@ static void SaveState()
     if (romName)
     {
         char* fileName = odroid_util_GetFileName(romName);
-        if (!fileName) abort();
+        if (!fileName)
```
Comment (OtherCrashOverride) [inline gnuboy-go/main/main.c]: This is not a file error. It is a memory allocation error for the string.

File: gnuboy-go/main/main.c [hunk 2]
```diff
@@ -310,7 +314,11 @@ static void LoadState(const char* cartName)
...
```
Comment (OtherCrashOverride) [inline gnuboy-go/main/main.c]: Also not a file error.

File: nesemu-go/main/main.c [hunk 3]
```diff
@@ -50,17 +51,21 @@ int app_main(void)
...
-		free(romName);
-	}
-	else
-	{
-		fileName = "nesemu-show3.nes";
-	}
+        if (romName)
+        {
+            fileName = odroid_util_GetFileName(romName);
+            if (!fileName)
```
Comment (OtherCrashOverride) [inline nesemu-go/main/main.c]: Also not a file error.

File: odroid-go-common/components/odroid/odroid_display.c [hunk 4]
```diff
@@ -1300,10 +1302,31 @@ void odroid_display_drain_spi()
...
+    switch(errNum)
+    {
+	case ODROID_SD_ERR_BADFILE:
+            ili9341_write_frame_rectangleLE(0, 0, image_sd_card_unknown.width, image_sd_card_unknown.height, image_sd_card_unknown.pixel_data); // Bad File image
+            break;
+	case ODROID_SD_ERR_NOCARD:
+            ili9341_write_frame_rectangleLE(0, 0, image_sd_card_alert.width, image_sd_card_alert.height, image_sd_card_alert.pixel_data); // No Card image
+            break;
+        default:
+            ili9341_write_frame_rectangleLE(0, 0, image_sd_card_alert.width, image_sd_card_alert.height, image_sd_card_alert.pixel_data); // Default error image
```
Comment (OtherCrashOverride) [inline odroid-go-common/components/odroid/odroid_display.c]: Do not display anything here.  Just abort.

File: odroid-go-common/components/odroid/odroid_display.h [hunk 5]
```diff
@@ -2,6 +2,9 @@
 
 #include <stdint.h>
 
+#define ODROID_SD_ERR_BADFILE 0x01
+#define ODROID_SD_ERR_NOCARD 0x02
```
Comment (OtherCrashOverride) [inline odroid-go-common/components/odroid/odroid_display.h]: Change these to an enum.

File: smsplus-go/main/main.c [hunk 6]
```diff
@@ -119,7 +119,11 @@ static void SaveState()
     if (romName)
     {
         char* fileName = odroid_util_GetFileName(romName);
-        if (!fileName) abort();
+        if (!fileName)
```
Comment (OtherCrashOverride) [inline smsplus-go/main/main.c]: Not a file error.

File: smsplus-go/main/main.c [hunk 7]
```diff
@@ -175,7 +179,11 @@ static void LoadState(const char* cartName)
...
```
Comment (OtherCrashOverride) [inline smsplus-go/main/main.c]: Not a file error.

File: odroid-go-common/sdkconfig [hunk 8]
```diff
@@ -20,7 +20,10 @@ CONFIG_LOG_BOOTLOADER_LEVEL_INFO=
 CONFIG_LOG_BOOTLOADER_LEVEL_DEBUG=
```
Comment (OtherCrashOverride) [inline odroid-go-common/sdkconfig]: This entire file needs to be omitted from the commit.

Comment (kamotswind) [reply odroid-go-common/sdkconfig]: Ah crap, I forgot about this. I'm not sure why the toolchain keeps wanting to rewrite it.

Comment (kamotswind) [reply odroid-go-common/sdkconfig]: I reverted it to your last commit.

**Comment by kamotswind** [comment]
This should be good now.

**Pull request merged**